### PR TITLE
Fix disabling shift+drag

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Address #1700:
shfit + drag should also be disabled if translation is a PROTO paramter.

Besides this, I found that a wrong commit was introduced here (PR https://github.com/cyberbotics/webots-dev/pull/6684): https://github.com/cyberbotics/webots-dev/pull/6684/commits/130676215fc06520be777b49afe8547c66f44b86
Probably, it was not clear why the manipulator could be NULL. But this commit breaks the mechanism caching the computation of valid translation and rotations fields. I will test if I can reproduce some errors after reverting this change.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:heavy_check_mark: (travis_unit)

File: src/webots/nodes/WbAbstractTransform.cpp [hunk 1]
```diff
@@ -257,11 +259,25 @@ bool WbAbstractTransform::isTranslationFieldVisible() const {
...
+    mCanBeRotated =
+      !WbNodeUtilities::isTemplateRegeneratorField(mBaseNode->findField("rotation", true)) && mIsRotationFieldVisible;
     mIsRotationFieldVisibleReady = true;
   }
   return mIsRotationFieldVisible;
 }
 
+bool WbAbstractTransform::canBeTranslated() const {
+  if (!mIsTranslationFieldVisibleReady)
+    isTranslationFieldVisible();
```
Comment (omichel) [inline src/webots/nodes/WbAbstractTransform.cpp:271]: This seems useless. Did you mean `return isTranslationFieldVisible();`?

Comment (stefaniapedrazzi) [reply src/webots/nodes/WbAbstractTransform.cpp:271]: No it is not useless.
The value `mCanBeTranslated` is computed and cached in `isTranslationFieldVisible()`.
So, if the value of `mCanBeTranslated` has not been computed yet, we need to call `isTranslationFieldVisible()` to compute it.

Comment (omichel) [reply src/webots/nodes/WbAbstractTransform.cpp:271]: I see. I was fooled by those mutable fields.

**Comment by stefaniapedrazzi** [comment]
I solved the conflicts in the ChangeLog

**Comment by stefaniapedrazzi** [comment]
Mutable fields could maybe not be very intuitive, but helps a lot for caching functionalities that will speed up the code.
I will now propose a change where the getter is not called, and the code will be slightly less efficient but should not affect the performance.

**Comment by stefaniapedrazzi** [comment]
For removing the mutable properties of cache fields, we have to modify a lot functions and remove the constant attribute of the node argument.
From my point of view, this will be worse than using mutable fields for cached values.

**Comment by omichel** [comment]
I don't say we must remove all these mutable properties. I just have the feeling that calling a getter, ignoring it's return value and expecting it modifies the object, looks to be a bad design. At minimum it deserves some comment to explain this weird behavior. At best some re-design may be helpful. See my previous comment about it.

**Comment by stefaniapedrazzi** [comment]
Here is a proposed version that should be more readable.

**Pull request merged**